### ISSUE TITLE
GameGuru MAX - Request - Behaviour for Moving/Animating Materials

### ISSUE THREAD

**Issue by ArgentArts** [TheGameCreators/GameGuruRepo#2554]
I would love to see a Behaviour added that would allow us to move materials along a surface. This would be good for simulating things like lava flows, water slowly moving, and other special effects. Being able to actually move or "slide" a material along a surface would allow for a nice, smooth motion as compared to using only a few "frames" from a sprite sheet.

**Comment by Mt-Chiliad-UFO** [comment]
Absolutely. I had this thought long ago, but haven't delved deeply into the concept. My idea then was to use particles and add that behavior to them. Since I couldn't change the foundation of Particles behavior as required, I didn't pursue it. Regardless of what you use, I do think it would only be a few lines of Lua code.
But... the discussion would have to go along the lines of "What is a material? - decals?" and "What would be the hit on fps" in assigning several dozens of "material objects" to Lua code? In Classic assigning even a relatively simple script to some dozens of objects certainly begins to effect frame rates. Been there, done that. But nonetheless, I fully agree with the idea. I could do some testing on this in Classic first...

**Comment by ArgentArts** [comment]
It seems to be a fairly common way to do things in other engines. 

> the discussion would have to go along the lines of "What is a material? - decals?"

A material is ... a material according to the engine. A texture applied to a surface of some kind. It's what is displayed. When you work in Unreal or Unity, there's not really a discussion as to what is a material. You open the material editor and you edit your material. If you want that material animated, there's a method to do it. It doesn't matter what that material is applied to ... as far as the engine's concerned, it's a material. ;)

As far as the hit to FPS - as with anything, that's up to the end-user/developer to decide if they want to use the feature. For example, I was playing with a really cool particle effect created with Particle Editor to create ground level moving fog in my sci-fi corridor. It looked great. But there were times it dragged my FPS down by half! I had to make a decision. Ultimately, I abandoned it. I may revisit it in the future to see if I can optimize it, but frame rate won out over the effect. ;)
